https://github.com/apitrace/apitrace/commit/d28a980802ad48568c87da02d630c8babfe163bb.patch
https://bugs.gentoo.org/812155

(rebased by sam)

From: Jose Fonseca <jfonseca@vmware.com>
Date: Wed, 1 Sep 2021 16:34:54 +0100
Subject: [PATCH] gltrace: Avoid __libc_dlsym and __libc_dlopen_mode on GLIBC
 2.34.

These GLIBC_PRIVATE symbols are gone from GLIBC 2.34 due to the merge of
libdl.so onto libc.so.

This means apitrace can't defend against infinite recursion when
used with Steam Overlay, but at least it should work otherwise.

Fixes https://github.com/apitrace/apitrace/issues/756
--- a/wrappers/dlsym.cpp
+++ b/wrappers/dlsym.cpp
@@ -34,7 +34,7 @@
 #include "os.hpp"
 
 
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ < 34
 
 
 #include <dlfcn.h>

